File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3097_shortest-subarray-with-or-at-least-k-ii/solutions/java/solution.java
class Solution {
    public int minimumSubarrayLength(int[] nums, int k) {
        int minLen = Integer.MAX_VALUE;
        int currentOR = 0;
        int left = 0;
        
        for (int right = 0; right < nums.length; right++) {
            currentOR |= nums[right];
            
            while (currentOR >= k) {
                minLen = Math.min(minLen, right - left + 1);
                currentOR &= ~nums[left];
                left++;
            }
        }
        
        return minLen == Integer.MAX_VALUE ? -1 : minLen;
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3097_shortest-subarray-with-or-at-least-k-ii/solutions/cpp/solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int minimumSubarrayLength(vector<int>& nums, int k) {
        int n = nums.size();
        int minLen = n + 1;
        int orSum = 0;
        int left = 0;
        
        for (int right = 0; right < n; ++right) {
            orSum |= nums[right];
            while (orSum >= k) {
                minLen = min(minLen, right - left + 1);
                orSum &= ~nums[left]; // Remove nums[left] from the current OR
                left++;
                if (left <= right) {  // Recalculate OR from left to right
                    orSum = nums[left];
                    for (int j = left + 1; j <= right; ++j) {
                        orSum |= nums[j];
                    }
                } else {
                    orSum = 0;  // Reset orSum since left passed right
                }
            }
        }
        
        return minLen == n + 1 ? -1 : minLen;
    }
};
